### ISSUE TITLE
Mac cOS compilation with Homebrew LLVM 18

### ISSUE THREAD

**Issue by w-bruns** [Normaliz/Normaliz#421]
It seems that LLVM 18 from Homebrew has severe problems resulting in `Trace/BPT trap: 5` crashes of normaliz.  Therefore we returned to MacOS 12 in the GitHub workflows.

**Comment by mahrud** [comment]
For what it's worth, I was able to bottle Normaliz and libnormaliz for macOS 14 without any issues: https://github.com/Macaulay2/homebrew-tap/pull/227

**Comment by w-bruns** [comment]
I don't have  Mac at my disposal. So my only access to MacOS is on GitHub. There are actually two problems that I have observed with the current MacOS-latest. Have a look at the commits on May 18 and 19.

1) Even the builds that use Apple LLVM failed because some tests that use fixed precision arithmetic seem to give wrong results. Actually these results are not uniquely determined by the input data, but depend on the workflow in the algorithm, and in particular on the parallelization. To safeguard against such irrelevant errors, the diff that compares the output to a reference file, excludes certain lines(unless we asked for STRICT checking): in `Makefile.classic` in `test` we see
```
ifeq ($(STRICT),yes)
DIFF = diff -b
else
DIFF = diff -b -I ".*embedding.*\|.*resulting.*\|.*triangulation.*\|.*float.*\|.*fixed.*"
#DIFF = diff -b
endif
```
I have no idea why this does not work with MacOS-latest. Is there a difference between GNU diff and MacOS diff?

2) The two builds that use LLVM 18 from Homebrew produce code running into `Trace/BPT trap: 5`. Perhaps this has to do with OpenMP, but OpenMP is the only reason to use the Homebrew compiller.

**Comment by w-bruns** [comment]
@linuszip

My grandson https://github.com/linuszip has analyzed the diff problem (1. in the preceding comment). It seems that the `diffutils` of Homebrew are installed implicitly with the other packages asked for in MacOS 12 and older, but no longer with MacOS 14 on GitHub. Installing them solves the problem.

**Comment by mahrud** [comment]
> 2\. The two builds that use LLVM 18 from Homebrew produce code running into `Trace/BPT trap: 5`. Perhaps this has to do with OpenMP, but OpenMP is the only reason to use the Homebrew compiller.

LLVM should of course work, but why not get Xcode working, too? As I mentioned four years ago, the file ax_openmp.m4 is outdated (2.59 instead of 2.69), but to make it work with Xcode you need an additional patch included in #355 that adds `-Xpreprocessor -fopenmp` to the test flags.

The homebrew formula for Normaliz has been using the following variable to work around this issue and compile using Xcode:
```
      ENV["OPENMP_CXXFLAGS"] = "-Xpreprocessor -fopenmp -I#{libomp.opt_include} #{libomp.opt_lib}/libomp.dylib"
```

**Comment by w-bruns** [comment]
Strangely, macos-15 (and LLVM 19) shows no problem at all. Therefore I assume that there is some incompatibility n between GitHub/macos-14 and LLVM 18.
